components/core/tests/test-NetworkReader.cpp

```diff
@@ -188,3 +208,47 @@ TEST_CASE("network_reader_illegal_offset", "[NetworkReader]") {
     size_t pos{};
     REQUIRE((clp::ErrorCode_Failure == reader.try_get_pos(pos)));
 }
+
+TEST_CASE("network_reader_with_http_header_kv_pairs", "[NetworkReader]") {
+    std::unordered_map<std::string, std::string> regular_http_header_kv_pairs;
+    // We use httpbin (https://httpbin.org/) to test the custom headers. This request will return a
+    // JSON object that contains the custom headers. We check if the headers are in the response.
+    constexpr int cNumRegularTestHeaders{10};
+    for (size_t i{0}; i < cNumRegularTestHeaders; i++) {
+        regular_http_header_kv_pairs.emplace(
+                fmt::format("Unit-Test-Key{}", i),
+                fmt::format("Unit-Test-Value{}", i)
+        );
+    }
+    clp::NetworkReader regular_reader{
+            "https://httpbin.org/headers",
+            0,
+            false,
+            clp::CurlDownloadHandler::cDefaultOverallTimeout,
+            clp::CurlDownloadHandler::cDefaultConnectionTimeout,
+            clp::NetworkReader::cDefaultBufferPoolSize,
+            clp::NetworkReader::cDefaultBufferSize,
+            regular_http_header_kv_pairs
+    };
+    auto content{nlohmann::json::parse(get_content(regular_reader))};
+    if (content.is_array() && 1 == content.size()) {
+        content = content.at(0);
+    }
+    auto const& headers{content.at("headers")};
+    REQUIRE(assert_curl_error_code(CURLE_OK, regular_reader));
+    for (int i = 0; i < cNumRegularTestHeaders; i++) {
```

Comment:
If applying the comment above, this should be replaced.
But as I commented in the last round of review, we shouldn't use `int` unless the number can be negative; we should use `{}` for typed variable initialization; and we should use `++i`  in general loop variable increment